### PR TITLE
feat: Session Reset

### PR DESCRIPTION
**Pull request by rhysbarrett**
### JIRA

[Reset session upon reaching the confirmation page](https://companieshouse.atlassian.net/browse/SR-113)

### Change Description

Updates Suppression Session model to allow session to be reset at the end of a successful submission, while allowing access to the Confirmation Page to continue up until either:
a) The user's session expires; or
b) The user begins a new submission (triggered by a successful POST on the Applicant Details page)

### Work Checklist

- [x] Tests added where applicable
- [x] Test coverage of new code meets target of 80%

### CODE REVIEW

File: src/middleware/NavigationMiddleware.ts [hunk 1]
```diff
@@ -13,17 +19,28 @@ export function NavigationMiddleware(): RequestHandler {
     const exempt: boolean = exemptions.some(exemption => urlMatches(exemption, url));
 
     if (exempt) {
+      loggerInstance().info(`${NavigationMiddleware.name} - ${url} exempt from navigation protection. Passing through.`);
```
[the remark this answers is not in the document]
Comment (rhysbarrett) [reply src/middleware/NavigationMiddleware.ts:22]: Nice. I notice that when I use this, the url is now included in the log body by default. Shall I remove the url from the log message for brevity?

File: src/models/SuppressionSessionModel.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ export interface SuppressionSession {
   applicationReference: string;
   navigationPermissions: string [];
   paymentDetails?: PaymentDetails;
+  previousApplicationReference?: string;
```
[the remark this answers is not in the document]
Comment (rhysbarrett) [reply src/models/SuppressionSessionModel.ts:7]: Ah much better.

File: src/middleware/NavigationMiddleware.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (n-r-b) [reply src/middleware/NavigationMiddleware.ts]: yes, why not :-)

File: src/middleware/NavigationMiddleware.ts [hunk 3]
```diff
@@ -37,12 +37,16 @@ export function NavigationMiddleware(): RequestHandler {
       loggerInstance().info(`${NavigationMiddleware.name} - user has no navigation permissions. Redirecting to service start.`);
       return res.redirect(APPLICANT_DETAILS_PAGE_URI);
 
-    } else if (!navigationPermissions.includes(url)) {
-      const redirectUrl: string = navigationPermissions[navigationPermissions.length - 1]
-      loggerInstance().info(`${NavigationMiddleware.name} - user has insufficient permissions for ${url}. Redirecting back to ${redirectUrl}.`);
-      return res.redirect(redirectUrl);
+    } else {
```
Comment (n-r-b) [inline src/middleware/NavigationMiddleware.ts:40]: can we break out of this `else`? i don't think it's needed

Comment (rhysbarrett) [reply src/middleware/NavigationMiddleware.ts:40]: Sure. Done the same for the `else` on line 40 due to preceding block's `return` statement.

Comment (n-r-b) [reply src/middleware/NavigationMiddleware.ts]: much better

**Review by n-r-b — APPROVED** [review]

**Comment by rhysbarrett** [comment]
Diff overcomplicates the most recent commit - I've moved each of the iterative tests into the same `for` loop instead of each test case having its own loop.

**Review by n-r-b — APPROVED** [review]

**Pull request merged**